src/fountain.rs

```diff
@@ -423,18 +432,18 @@ impl Decoder {
     /// ```
     #[must_use]
     pub fn validate(&self, part: &Part) -> bool {
-        if part.sequence_count != self.sequence_count {
-            return false;
-        }
-        if part.message_length != self.message_length {
-            return false;
-        }
-        if part.checksum != self.checksum {
+        if self.received.is_empty() {
             return false;
         }
-        if part.data.len() != self.fragment_length {
+
+        if part.sequence_count != self.sequence_count
```

Comment:
Can you explain the intention with this diff? This is only reshuffling stuff, right?

If you want to keep it, can you please put it into a separate commit, because it seems unrelated to the commit message (unless I didn't get something).

Reply:
Just to keep the same formatting between the if blocks as the if/returns block would have been much bigger when adding the other equalities, so that's the reason for the same commit.

Reply:
Added an extra commit for this one and for the check